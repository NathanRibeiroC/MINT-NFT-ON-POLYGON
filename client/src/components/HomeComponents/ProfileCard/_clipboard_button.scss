$back-color: #bcbcbc;
$check-color: #2C974B;

.clipboard-button{
    width: 20px;
    height: 20px;
    display: flex;
    align-items: center;
    border-radius: 4px;
    border-width: 0;
    background-color: #e2e2e2;
    border: solid 1px $back-color;
    padding: 2px;
    box-shadow: inset 0 0 1.6em -0.6em rgba(0, 0, 0, 0.15);

    &.content-copied{
        border: solid 1px $check-color;
    }

    &:hover{
        background-color: #d5d3d3;

        .tooltiptext{

            visibility: visible;
            opacity: 1;

            &:hover{
                visibility: hidden;
                opacity: 0;
            }
        }
    }

    .tooltiptext{
        visibility: hidden;
        width: 50px;
        background-color: #555;
        font-size: 10px;
        color: #fff;
        text-align: center;
        border-radius: 6px;
        padding: 5px;
        position: relative;
        z-index: 200;
        bottom: 250%;
        left: 320%;
        margin-left: -75px;
        opacity: 0;
        transition: opacity 0.3s;

        &:after{
            content: "";
            position: absolute;
            top: 100%;
            left: 60%;
            margin-left: -5px;
            border-width: 5px;
            border-style: solid;
            border-color: #555 transparent transparent transparent;
        }

        &.copied{
            visibility: visible;
            bottom: 200%;
            opacity: 1;

            &:hover{
                visibility: visible;
                opacity: 1;
            }
        }
    }
}

.clipboard-icon{
    width: 13.5px;
    height: 13.5px;
    left: 110%;
    bottom: -0%;
    position: relative;
}

.check-icon{
    width: 13.5px;
    height: 13.5px;
    left: 116%;
    bottom: -0%;
    position: relative;
}