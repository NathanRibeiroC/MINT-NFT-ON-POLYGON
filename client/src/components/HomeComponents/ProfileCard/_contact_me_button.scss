$back-color: #bcbcbc;

  .cssbuttons-io-button {
    border: none;
    margin-left: 10px;
    background: #5e7791;
    color: white;
    width: 93%;
    font-family: "Jost", sans-serif;
    padding: 0.35em;
    padding-left: 6em;
    font-size: 14px;
    font-weight: 500;
    border-radius: 6px;
    letter-spacing: 0.05em;
    display: flex;
    align-items: center;
    box-shadow: inset 0 0 1.6em -0.6em rgba(0, 0, 0, 0.15);
    overflow: hidden;
    position: relative;
    height: 3em;
    padding-right: 3.3em;

    .icon {
      background: white;
      margin-left: 1em;
      position: absolute;
      display: flex;
      align-items: center;
      justify-content: center;
      height: 2.4em;
      width: 2.6em;
      border-radius: 5px;
      box-shadow: 0.1em 0.1em 0.6em 0.2em rgba(0, 0, 0, 0.15);
      right: 0.3em;
      transition: all 0.3s;

      .contact-button-icon {
        height: 65%;
        width: 65%;
        transition: transform 0.3s;
      }

    }

    &:hover{
      .icon {
        width: calc(100% - 0.6em);
      }

      .contact-button-icon{
        transform: translateX(0.1em) rotate(45deg);
      }
    }

    &:active{
      .icon {
        transform: scale(0.95);
      }
    }
  }