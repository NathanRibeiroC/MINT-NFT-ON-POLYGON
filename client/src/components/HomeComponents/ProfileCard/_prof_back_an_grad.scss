// ABOUT TAB
.profile-back-grad-about-tab{
    position: absolute;
    background: linear-gradient(-90deg, #ee7752, #e73c7e, #23a6d5, #23d5ab);
	background-size: 400% 400%;
	animation: gradient 15s ease infinite;
	height: 200px;
    width: 300px;
	z-index: 100;
	border-radius: 8px;
}

// EXPERIENCE TAB
.profile-back-grad-experience-tab{
	position: absolute;
    background: linear-gradient(-90deg, #ee7752, #e73c7e, #23a6d5, #23d5ab);
	background-size: 400% 400%;
	animation: gradient 15s ease infinite;
	height: 100px;
    width: 300px;
	z-index: 100;
	border-radius: 8px;
}

@keyframes gradient {
	0% {
		background-position: 0% 50%;
	}
	50% {
		background-position: 100% 50%;
	}
	100% {
		background-position: 0% 50%;
	}
}