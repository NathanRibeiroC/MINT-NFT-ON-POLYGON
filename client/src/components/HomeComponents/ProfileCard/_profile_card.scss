@import url("https://fonts.googleapis.com/css?family=DM+Sans:400,500|Jost:400,500,600&display=swap");

$back-color: white;
$accent: #8855ff;

$card-width: 300px;

//GENERAL WINDOW (NAVBAR + CARD)

nav {
  background: #fdfdfd;
  padding: 0;
  border-radius: 10px;
  border-top-left-radius: 0;
  border-top-right-radius: 0;
  border-top: 1px solid #eeeeee;
  height: 44px;
  z-index: 104;
}

.card-about {
  font-family: "Jost", sans-serif;
  color: #2b2c48;
  width: $card-width;
  height: 410px;
  border-radius: 10px;
  background: #fdfdfd;
  overflow: hidden;
  box-shadow: 0 0 0 8px rgba(255, 255, 255, 0.2);
  display: flex;
  flex-direction: column;
}

ul,
li {
  list-style: none;
  padding: 0;
  margin: 0;
  font-family: "Poppins", sans-serif;
  font-weight: 500;
  font-size: 14px;
}

ul {
  display: flex;
  width: 100%;
}

li {
  width: 100%;
  padding: 10px 15px;
  position: relative;
  background: $back-color;
  cursor: pointer;
  height: 24px;
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  user-select: none;
}

li.selected {
  background: #eee;
  z-index: 104;
}



.underline {
  width: 100%;
  height: 2px;
  background: $accent;
  position: absolute;
  z-index: 103;
  top: -1px;
}

//ABOU TAB
.about-tab {
  width: 100%;
  position: relative;
  display: flex;
  flex-direction: column;
  transition: 0.3s;
  box-shadow: 0 0 0 8px rgba(255, 255, 255, 0.2);
  align-items: center;
  padding-top: 30px;
  z-index: 103;

  .profile-picture{
    border-radius: 100%;
    height: 100px;
    width: 100px;
    box-shadow: 0 8px 8px rgba(0, 0, 0, 0.2);
    margin-bottom: 6px;
    left: 0%;
  }

  h1{
    font-size: 22px;
    margin-bottom: 0;
    padding-bottom: 5px;
    margin-top: -0.1px;
  }

  h2{
    margin: 0;
    padding: 0;
    font-size: 11px;
    letter-spacing: 1px;
    padding: 0 0 20px 0;
    font-weight: 500;
    color: #918c8a;
    padding-bottom: 25px;
  }

  .main-text-about{
    padding-left: 10px;
    background-color: $back-color;
    animation: fadeIn 0.6s both;
    h3{
      font-size: 13px;
      font-weight: 700;
      margin: 0;
      padding: 0 0 8px 0;
    }

    p{
      font-size: 13px;
      color: #6f6f7b;
      line-height: 1.5;
      font-family: "DM Sans", sans-serif;
      margin: 0;
      padding-right: 17px;
      padding-bottom: 17px;
    }
  }

  .social-media-card{
    padding-right:225px;
    padding-bottom: 30px;
    background-color: $back-color;
    margin: 0;
    animation: fadeIn 0.6s both;
    .github-icon{
      padding-right: 5px;
      height: 25px;
      cursor: pointer;
      transition: 0.3s;

      &:hover{
        fill:  darken(rgb(165, 181, 206), 20%);
      }
    }

    .linkedin-icon{
      height: 25px;
      cursor: pointer;

      &:hover{
        fill: darken(rgb(165, 181, 206), 20%);
      }
    }
  }
}

//waves animation
//about tab
.waves-about-tab {
  position:absolute;
  top: 200px;
  z-index: 102;
  width: 300px;
  height:1000px;
  margin-bottom:0; /*Fix for safari gap*/
  min-height:50px;
  max-height:60px;
}

.aux-wave-container-about-tab{
  position: absolute;
  z-index: 102;
  width: 300px;
  height: 60px;
  top: 260px;
  margin: 0;
  padding: 0;
  background-color: $back-color;
}

//exp tab
.waves-exp-tab {
  position:absolute;
  top: 120px;
  z-index: 102;
  width: 300px;
  margin-bottom:0; /*Fix for safari gap*/
  min-height:50px;
  max-height:60px;
}

//exp tab

.aux-wave-container-exp-tab{
  position: absolute;
  z-index: 102;
  width: 300px;
  height: 7px;
  top: 175px;
  margin: 0;
  padding: 0;
  background-color: $back-color;
}

// WAVE ANIMATION

.parallax1 > use{
  animation: move-forever 25s cubic-bezier(.55,.5,.45,.5) infinite;
  animation-delay: -2s;
  animation-duration: 7s;
}

.parallax2 > use{
  animation: move-forever 25s cubic-bezier(.55,.5,.45,.5) infinite;
  animation-delay: -3s;
  animation-duration: 10s;
}

.parallax3 > use {
  animation: move-forever 25s cubic-bezier(.55,.5,.45,.5) infinite;
  animation-delay: -4s;
  animation-duration: 13s;
  -moz-animation-delay: -4s;
  -moz-animation-duration: 13s;
  -webkit-animation-delay: -4s;
  -webkit-animation-duration: 13s;
  -o-animation-delay: -4s;
  -o-animation-duration: 13s;
  -ms-animation-delay: -4s;
  -ms-animation-duration: 13s;
}
.parallax4 > use {
  animation: move-forever 25s cubic-bezier(.55,.5,.45,.5) infinite;
  animation-delay: -5s;
  animation-duration: 20s;
  -moz-animation-delay: -5s;
  -moz-animation-duration: 20s;
  -webkit-animation-delay: -5s;
  -webkit-animation-duration: 20s;
  -o-animation-delay: -5s;
  -o-animation-duration: 20s;
  -ms-animation-delay: -5s;
  -ms-animation-duration: 20s;
}

@keyframes move-forever {
  0% {
   transform: translate3d(-90px,0,0);
   -moz-transform: translate3d(-90px,0,0);
   -webkit-transform: translate3d(-90px,0,0);
   -o-transform: translate3d(-90px,0,0);
   -ms-transform: translate3d(-90px,0,0);
  }
  100% { 
    transform: translate3d(85px,0,0);
    -moz-transform: translate3d(85px,0,0);
    -webkit-transform: translate3d(85px,0,0);
    -o-transform: translate3d(85px,0,0);
    -ms-transform: translate3d(85px,0,0);
  }
}

//EXPERIENCE TAB

.card-experience{
  font-family: "Jost", sans-serif;
  width: $card-width;
  height: 550px;
  border-radius: 10px;
  background: $back-color;
  overflow: hidden;
  box-shadow: 0 0 0 8px rgba(255, 255, 255, 0.2);
}

.profile-picture-small-size{
  position: relative;
  width: 50px;
  height: 50px;
  border-radius: 100%;
  box-shadow: 0 3px 3px rgba(0, 0, 0, 0.2);
  margin-right: 15px;
  z-index: 200;
  transform: none;
  left: 0%;
}

.profile-tex-small-size{
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;

  h1{
    position: relative;
    font-size: 19px;
    color: #2b2c48;
    margin: 0;
    padding: 0;
    z-index: 103;
  }

  h2{
    position: relative;
    font-size: 10px;
    letter-spacing: 1px;
    font-weight: 500;
    color: #918c8a;
    margin: 0;
    z-index: 103;
  }
}

.profile-div-small-size{
  display: flex;
  padding-top: 20px;
  padding-left: 20px;
}

.experience-tab{
  height: 508px;
}

.content-small-tab{
  h3{
    font-family: "Poppins", sans-serif;
    z-index: 103;
    font-size: 12px;
    font-weight: 700;
    color: #2b2c48;
    position: absolute;
    padding-top: 10px;
    padding-left: 20px;
    animation: fadeIn 0.6s both;
  }
}

.card-content{
  padding-top: 57px;
  padding-left: 38px;
}

.card-timeline {
  position: relative;
  width: 280px;
  height: 370px;
  &:after {
    background: linear-gradient(
      to top,
      rgba(134, 214, 243, 0) 0%,
      rgba(81, 106, 204, 1) 100%
    );
    content: "";
    left: 41px;
    width: 2px;
    top: 1.5px;
    height: 100%;
    position: absolute;
    content: "";
  }
}

.card-item {
  position: relative;
  padding-left: 60px;
  padding-right: 20px;
  padding-bottom: 30px;
  z-index: 103;
  &:last-child {
    padding-bottom: 5px;
  }

  &:after {
    content: attr(data-year);
    position: absolute;
    top: 0;
    left: 36px;
    width: 8px;
    height: 8px;
    line-height: 0.6;
    border: 2px solid #fff;
    font-size: 10.5px;
    text-indent: -60px;
    border-radius: 50%;
    color: rgba(#868686, 0.7);
    background: linear-gradient(
      to bottom,
      lighten(#516acc, 20%) 0%,
      #516acc 100%
    );
  }
}

.card-item-title {
  font-weight: 500;
  font-size: 13px;
  padding-bottom: 7px;
}

.card-item-desc {
  padding-top: 7px;
  font-size: 12px;
  color: #6f6f7b;
  line-height: 1.5;
  font-family: "DM Sans", sans-serif;
  width: 150px;

  &:after {
    font-family: "Jost", sans-serif;
    font-weight: 100;
    content: attr(data-year);
    position: absolute;
    top: 15px;
    left: 38px;
    width: 8px;
    height: 8px;
    line-height: 0.6;
    font-size: 10.5px;
    text-indent: -60px;
    border-radius: 50%;
    color: rgba(#868686, 0.7);
  }
}

.card-section {
  display: none;
  display: block;
  animation: fadeIn 0.6s both;
}

@keyframes fadeIn {
  0% {
    opacity: 0;
    transform: translatey(40px);
  }
  100% {
    opacity: 1;
  }
}

//CONTACT TAB

.card-contact{
  font-family: "Jost", sans-serif;
  color: #2b2c48;
  width: $card-width;
  height: 435px;
  border-radius: 10px;
  background-color: $back-color;
  overflow: hidden;
  box-shadow: 0 0 0 8px rgba(255, 255, 255, 0.2);
  display: flex;
  flex-direction: column;
  // position: absolute;
  // perspective: 500px;

  // &:hover .content{
  //   transform: rotateY( 180deg ) ;
  //   transition: transform 0.5s;
  // }
}

.contact-tab{
  padding-bottom: 21px;
}

.contact-data{
  position: relative;
  margin-top: 50px;
  font-family: "DM Sans", sans-serif;
  font-size: 13px;
  color: #6f6f7b;
  animation: fadeIn 0.6s both;
  .contact-single-data{
    display: flex;
    margin-bottom: 20px;
    align-items: center;

    &.email2{
      margin-bottom: 28px;
    }
  }

  .data-text{
    margin-right: 25px;
    padding: 0;
    line-height: 5px;
    width: 180px;
    line-height: 1.3;
    -webkit-touch-callout: none;
    -webkit-user-select: none; 
    -khtml-user-select: none; 
    -moz-user-select: none; 
    -ms-user-select: none; 
    user-select: none; 
  }

  .phone-icon{
    height: 20px;
    width: 20px;
    padding-top: 1.5px;
    margin-right: 2.5px;
    margin-left: 15px;
  }

  .location-icon{
    height: 20px;
    width: 20px;
    padding-top: 1.5px;
    margin-right: 1px;
    margin-left: 15px;
  }

  .email-icon{
    height: 20px;
    width: 20px;
    padding-top: 1.5px;
    margin-right: 1px;
    margin-left: 15px;
  }
  
  .data-div{
    border-right: 1px solid #dfe2ec;
    height: 27px;
    margin-right: 10px;
    margin-left: 5px;
    color: $back-color;
    -webkit-user-select: none;
    -khtml-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    -o-user-select: none;
    user-select: none;

    &.location{
      margin-left: 7px;
    }

    &.phone{
      margin-left: 4px;
    }
  }
}

//FLIP CARD

.flip-card-wrapper {
    // width: 300px;
    // height: 390px;
    float: left;
    // perspective: 500px;
    // border-radius: 10px;
    // box-shadow: 0 0 0 8px rgba(255, 255, 255, 0.2);
}

.flip-card-content{
  height: 0;
  margin: 0;
  padding: 0;
  transition: transform 1s;
  transform-style: preserve-3d;
}

.back-card{
  transform: rotateY( -180deg );
  transition: transform 0.5s;
}


