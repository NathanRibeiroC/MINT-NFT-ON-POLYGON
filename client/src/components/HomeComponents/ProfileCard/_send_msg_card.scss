.card {
    width: 300px;
    height: 390px;
    float: left;
    perspective: 500px;
    border-radius: 10px;
    box-shadow: 0 0 0 8px rgba(255, 255, 255, 0.2);
  }
  
  .content {
    height: 0;
    margin: 0;
    padding: 0;
    box-shadow: 0 0 15px rgba(0,0,0,0.1);
    transition: transform 1s;
    transform-style: preserve-3d;
  }
  
//   .card:hover .content {
//     transform: rotateY( 180deg ) ;
//     transition: transform 0.5s;
//   }
  
  .front,
  .back {
    backface-visibility: visible;
  }
  
  .back {
    transform: rotateY( -180deg );
    transition: transform 0.5s;
  }