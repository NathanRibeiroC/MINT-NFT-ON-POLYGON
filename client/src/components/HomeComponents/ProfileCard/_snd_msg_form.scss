.snd-msg-form{
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: .5fr 1fr 1 fr .5fr;
}

.row-1{
    margin: .7em;
}

.row-2{
    display: flex;
    flex-direction: column;
    gap: .5em;
    margin: 0 1em 0 .7em;
}

.row-3{
    margin: .5em 1em 1em .7em;
}