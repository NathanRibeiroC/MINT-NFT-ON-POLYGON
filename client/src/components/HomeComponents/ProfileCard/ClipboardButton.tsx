import { useEffect, useState } from "react";
import { ReactComponent as ClipboardIcon } from "../../../../public/clipboard-icon.svg";
import { ReactComponent as CheckIcon } from "../../../../public/check-icon.svg";
import "./_clipboard_button.scss";
import isEmpty from "../../../utils/isEmpty";

interface PropsType {
  dataClassName: string;
}

const MILLISECONDS = 800;

function ClipboardButton({ dataClassName }: PropsType): JSX.Element {
  const [isCopiedToClipboard, setIsCopiedToClipboard] =
    useState<boolean>(false);
  const [triggerUseEffect, setTriggerUseEffect] = useState<boolean>(false);

  const copyToClipboard = () => {
    const classTextContent =
      document.getElementsByClassName(dataClassName)[0].textContent;

    if (!isEmpty(classTextContent)) {
      navigator.clipboard.writeText(classTextContent || "");
      setTriggerUseEffect(true);
      setIsCopiedToClipboard(true);
    }
  };

  const resetState = () => {
    setTriggerUseEffect(false);
    setIsCopiedToClipboard(false);
  };

  useEffect(() => {
    if (triggerUseEffect) {
      const timer = setTimeout(() => resetState(), MILLISECONDS);
      return () => {
        clearTimeout(timer);
      };
    }
    return undefined;
  }, [triggerUseEffect]);

  const copyToClipboardTextAdd = (
    <span className="tooltiptext">
      Copy to
      <br />
      clipboard
    </span>
  );

  const copiedToClipboardTextAdd = (
    <span className="tooltiptext copied">Copied !</span>
  );

  return (
    <button
      type="button"
      className={
        isCopiedToClipboard
          ? "clipboard-button content-copied"
          : "clipboard-button"
      }
      onClick={copyToClipboard}
    >
      {isCopiedToClipboard ? copiedToClipboardTextAdd : copyToClipboardTextAdd}
      {isCopiedToClipboard ? (
        <CheckIcon className="check-icon" />
      ) : (
        <ClipboardIcon className="clipboard-icon" />
      )}
    </button>
  );
}
export default ClipboardButton;
