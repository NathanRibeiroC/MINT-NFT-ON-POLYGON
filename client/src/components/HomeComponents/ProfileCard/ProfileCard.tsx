import './_profile_card.scss';
import { useState } from 'react';
import { motion } from 'framer-motion';
import AboutTab from './AboutTab';
import ExperienceTab from './ExperienceTab';
import ContactTab from './ContactTab';
import { useAppSelector } from '../../../redux/app/hooks';
import SendMessageCard from './SendMessage';

function ProfileCard(): JSX.Element {
  const tabs = [
    { label: 'ABOUT', card_class_name: 'card-about' },
    { label: 'EXPERIENCE', card_class_name: 'card-experience' },
    { label: 'CONTACT', card_class_name: 'card-contact' },
  ];

  const [selectedTab, setSelectedTab] = useState(tabs[0]);
  const flipcard = useAppSelector((state) => state.flipProfileCard.value);

  const cardTabs = (
    <div className="front-card">
      <div className={selectedTab.card_class_name}>
        <main>
          {selectedTab.label === 'ABOUT' ? <AboutTab /> : null}
          {selectedTab.label === 'EXPERIENCE' ? <ExperienceTab /> : null}
          {selectedTab.label === 'CONTACT' ? <ContactTab /> : null}
        </main>
        <nav className="card-navbar">
          <ul>
            {tabs.map((item) => (
              <li
                role="tab"
                key={item.label}
                className={item.label === selectedTab.label ? 'selected' : ''}
                onClick={() => setSelectedTab(item)}
                onKeyDown={() => setSelectedTab(item)}
              >
                {item.label === selectedTab.label ? (
                  <motion.div
                    className="underline"
                    layoutId="underline"
                    initial={{ opacity: 0, scale: 0.5 }}
                    animate={{ opacity: 1, scale: 1 }}
                    transition={{
                      duration: 0.8,
                      delay: 0.5,
                      ease: [0, 0.71, 0.2, 1.01],
                    }}
                  />
                ) : null}
                {`${item.label}`}
              </li>
            ))}
          </ul>
        </nav>
      </div>
    </div>
  );

  const sendMessageBackCard = (
    <div className="back-card">
      <SendMessageCard />
    </div>
  );

  return (
    <div className="flip-card-wrapper">
      <div className="flip-card-content">
        {!flipcard ? cardTabs : sendMessageBackCard}
      </div>
    </div>
  );
}

export default ProfileCard;
