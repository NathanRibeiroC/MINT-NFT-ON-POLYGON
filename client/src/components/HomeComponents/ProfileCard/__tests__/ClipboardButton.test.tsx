// app.test.js
// import {render, screen} from '@testing-library/react'
// import userEvent from '@testing-library/user-event'
// import '@testing-library/jest-dom'
// import App from '../../../../App'
// import React from 'react'
// import {BrowserRouter} from 'react-router-dom'
// import { describe, it, expect } from 'vitest';

// const renderWithRouter = (ui:any, {route = "/"} = {}) => {
//   window.history.pushState({}, 'Test page', route)

//   return {
//     user: userEvent.setup(),
//     ...render(ui, {wrapper: BrowserRouter}),
//   }
// }

// describe('Navbar Behavior', () => {
//   const navOptions = ["Services","Health & Wellness"];

//   it('Testing navbar expantion for Services option', async () => {

//     const {user} = renderWithRouter(App)

//     // expect(screen.getByText(navOptions[0])).toBeInTheDocument()
  
//     // await user.click(screen.getByText("CONTACT"))
  
//     expect(screen.getByText("ABOUT")).toBeInTheDocument()
//   })

//   // it('Testing navbar expantion for Health And Services option', async () => {

//   //   const {user} = renderWithRouter(<App />)

//   //   expect(screen.getByText(navOptions[1])).toBeInTheDocument()
  
//   //   await user.click(screen.getByText(navOptions[1]))
  
//   //   expect(screen.getByText("Exchange")).toBeInTheDocument()
//   // })
// });
import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';
import ProfileCard from '../ProfileCard';
import { Provider } from 'react-redux';
import isEmpty from '../../../../utils/isEmpty';
import ClipboardButton from '../ClipboardButton';
import renderer from 'react-test-renderer';
import GeneralFormInput from '../../../GeneralComponents/GeneralFormInput';
import { ErrorBoundary } from "react-error-boundary";

import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'
describe('App tests', () => {
  test('Link renders correctly', () => {
    const getInputValue = jest.fn(x => x);
    const tree = renderer
      .create(
        <GeneralFormInput id="fullName" labelName="Full Name" getInputValue={getInputValue}/>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();

  });

  test('loads and displays greeting', async () => {
    const getInputValue = jest.fn(x => x);
    // ARRANGE
    const { getByLabelText } = render(
        <GeneralFormInput id="fullName" labelName="Full Name" getInputValue={getInputValue}/>
    )
    const input = screen.getByLabelText('Full Name')
    fireEvent.change(input, {target: {value: '23'}})
    expect(getInputValue.mock.calls).toHaveLength(1);
  })
    
});
