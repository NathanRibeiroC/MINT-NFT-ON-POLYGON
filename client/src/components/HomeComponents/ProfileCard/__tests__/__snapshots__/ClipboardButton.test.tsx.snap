// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests Link renders correctly 1`] = `
<div
  className="gen-input-group"
>
  <input
    className="gen-form-input"
    id="fullName"
    onChange={[Function]}
    required={true}
    type="text"
  />
  <label
    className="user-label"
    htmlFor="fullName"
  >
    Full Name
  </label>
</div>
`;
